systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java

```diff
@@ -390,6 +396,97 @@ void testTlsExternalUser(ExtensionContext extensionContext) {
+
+        LOGGER.info("Verifying that Secret corresponding to KafkaUser: {}/{} has expected label: {}={}", namespace, testStorage.getUsername(), Labels.STRIMZI_CLUSTER_LABEL, userListeningClusterName);
+        TestUtils.waitFor("Wait for the Secret {}: {}/{} to exist while also having label to corresponding Kafka cluster", Constants.GLOBAL_POLL_INTERVAL, Constants.GLOBAL_TIMEOUT,
+            () -> kubeClient().listSecrets(namespace)
+                    .stream()
+                    .anyMatch(secret -> testStorage.getUsername().equals(secret.getMetadata().getName())
+                        && secret.getMetadata().getLabels().containsKey(Labels.STRIMZI_CLUSTER_LABEL)
+                        && secret.getMetadata().getLabels().get(Labels.STRIMZI_CLUSTER_LABEL).equals(userListeningClusterName)
+                    )
+        );
+
+        LOGGER.info("Verifying KafkaUser: {}/{} is mapped into Kafka resource user in Kafka: {}/{}", namespace, testStorage.getUsername(), namespace, userListeningClusterName);
+        TestUtils.waitFor("Wait for the KafkaUser CR mapping into a Kafka user resource", Constants.GLOBAL_POLL_INTERVAL, Constants.GLOBAL_TIMEOUT,
+            () -> {
+                String getUserResult = KafkaCmdClient.describeUserUsingPodCli(namespace, scraperPodName, KafkaResources.plainBootstrapAddress(userListeningClusterName), "CN=" + testStorage.getUsername());
+                return getUserResult.contains(testStorage.getUsername());
+            });
+
+        LOGGER.info("Verifying KafkaUser: {}/{} is not present in Kafka ecosystem in Kafka/{}", namespace, testStorage.getUsername(), userIgnoringClusterName);
+        String getUserResult = KafkaCmdClient.describeUserUsingPodCli(namespace, scraperPodName, KafkaResources.plainBootstrapAddress(userIgnoringClusterName), "CN=" + testStorage.getUsername());
+        assertThat("KafkaUser CR is not mapped as Kafka resource user", getUserResult, not(containsString(testStorage.getUsername())));
+
+        // Clean up KafkaUser and Kafka
+
+        LOGGER.info("Cleaning up KafkaUser: {}/{}", namespace, testStorage.getUsername());
+        try {
+            KafkaUserResource.kafkaUserClient().inNamespace(namespace).withName(user.getMetadata().getName()).withPropagationPolicy(DeletionPropagation.FOREGROUND).delete();
+            KafkaUserUtils.waitForKafkaUserDeletion(namespace, user.getMetadata().getName());
+        } catch (Exception e) {
+            LOGGER.error(e);
+        }
+
+        LOGGER.info("Cleaning up Kafka: {}/{}", namespace, userListeningClusterName);
+        try {
+            KafkaResource.kafkaClient().inNamespace(namespace).withName(userListeningClusterName).withPropagationPolicy(DeletionPropagation.FOREGROUND).delete();
+            KafkaUtils.waitForKafkaDeletion(namespace, userListeningClusterName);
+        } catch (Exception e) {
+            LOGGER.error(e);
+        }
```

Comment:
i failed to notice if this is common practise here, but as I am working in shared namespace i wanted to safely clean after it (and as I am not interested in testing deletions it is in try/catch blocks) WDYT ?

Reply:
ResourceManager should handle it. If no, it is probably a bug.

Reply:
thank you, removed.